zebra-scan/src/scan.rs

```diff
@@ -0,0 +1,58 @@
+//! The scan task.
+
+use std::time::Duration;
+
+use color_eyre::{eyre::eyre, Report};
+use tower::{buffer::Buffer, util::BoxService, Service, ServiceExt};
+use tracing::info;
+
+use crate::storage::Storage;
+
+type State = Buffer<
+    BoxService<zebra_state::Request, zebra_state::Response, zebra_state::BoxError>,
+    zebra_state::Request,
+>;
+
+/// Wait a few seconds at startup so tip height is always `Some`.
+const INITIAL_WAIT: Duration = Duration::from_secs(10);
+
+/// The amount of time between checking and starting new scans.
+const CHECK_INTERVAL: Duration = Duration::from_secs(10);
+
+/// Start the scan task given state and storage.
+///
+/// - This function is dummy at the moment. It just makes sure we can read the storage and the state.
+/// - Modificatiuons here might have an impact in the `scan_task_starts` test.
+/// - Real scanning code functionality will be added in the future here.
+pub async fn start(mut state: State, storage: Storage) -> Result<(), Report> {
+    // We want to make sure the state has a tip height available before we start scanning.
+    std::thread::sleep(INITIAL_WAIT);
+
+    loop {
+        // Make sure we can query the state
+        let request = state
+            .ready()
+            .await
+            .map_err(|e| eyre!(e))?
+            .call(zebra_state::Request::Tip)
+            .await
+            .map_err(|e| eyre!(e));
+
+        let tip_height = match request? {
+            zebra_state::Response::Tip(tip) => tip.unwrap().0,
+            _ => unreachable!("unmatched response to a state::Tip request"),
+        };
+
+        // Read keys from the storage
+        let available_keys = storage.get_sapling_keys();
+
+        for key in available_keys {
+            info!(
+                "Scanning the blockchain for key {} from block 1 to {}",
+                key.0, tip_height.0
+            );
+        }
```

Comment:
This `unwrap()` will cause a panic when the state is empty, which can happen on startup and during some tests. I'm not sure if that's the error you're having?

Since this is a just dummy code, we don't need to unwrap anything, we should just be able to print the whole tip:
```suggestion
        let tip = match request? {
            zebra_state::Response::Tip(tip) => tip,
            _ => unreachable!("unmatched response to a state::Tip request"),
        };

        // Read keys from the storage
        let available_keys = storage.get_sapling_keys();

        for key in available_keys {
            info!(
                "Scanning the blockchain for key {} from block 1 to {}",
                key.0, tip,
            );
        }
```

Reply:
Applied this at https://github.com/ZcashFoundation/zebra/pull/7989/commits/27d29364677a28bd572cd84a9261f5ef3876b8e8